gcloud-java-dns/src/test/java/com/google/gcloud/dns/ChangeRequestTest.java

```diff
@@ -0,0 +1,219 @@
+    ChangeRequest partial = ChangeRequest.builder().build();
+    assertEquals(partial, partial.toBuilder().build());
+    partial = ChangeRequest.builder().id(ID).build();
+    assertEquals(partial, partial.toBuilder().build());
+    partial = ChangeRequest.builder().add(RECORD1).build();
+    assertEquals(partial, partial.toBuilder().build());
+    partial = ChangeRequest.builder().delete(RECORD1).build();
+    assertEquals(partial, partial.toBuilder().build());
+    partial = ChangeRequest.builder().additions(ADDITIONS).build();
+    assertEquals(partial, partial.toBuilder().build());
+    partial = ChangeRequest.builder().deletions(DELETIONS).build();
+    assertEquals(partial, partial.toBuilder().build());
+    partial = ChangeRequest.builder().startTimeMillis(START_TIME_MILLIS).build();
+    assertEquals(partial, partial.toBuilder().build());
+    partial = ChangeRequest.builder().status(STATUS).build();
+    assertEquals(partial, partial.toBuilder().build());
+  }
+
+  @Test
+  public void testClearAdditions() {
+    ChangeRequest clone = CHANGE.toBuilder().clearAdditions().build();
+    assertTrue(clone.additions().isEmpty());
+    assertFalse(clone.deletions().isEmpty());
+  }
+
+  @Test
+  public void testAddAddition() {
+    try {
+      CHANGE.toBuilder().add(null).build();
+      fail("Should not be able to add null DnsRecord.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    ChangeRequest clone = CHANGE.toBuilder().add(RECORD1).build();
+    assertEquals(CHANGE.additions().size() + 1, clone.additions().size());
+  }
+
+  @Test
+  public void testAddDeletion() {
+    try {
+      ChangeRequest clone = CHANGE.toBuilder().delete(null).build();
+      fail("Should not be able to delete null DnsRecord.");
+    } catch (NullPointerException e) {
+      // expected
+    }
+    ChangeRequest clone = CHANGE.toBuilder().delete(RECORD1).build();
+    assertEquals(CHANGE.deletions().size() + 1, clone.deletions().size());
+  }
+
+  @Test
+  public void testClearDeletions() {
+    ChangeRequest clone = CHANGE.toBuilder().clearDeletions().build();
+    assertTrue(clone.deletions().isEmpty());
+    assertFalse(clone.additions().isEmpty());
+  }
+
+  @Test
+  public void testRemoveAddition() {
+    ChangeRequest clone = CHANGE.toBuilder().removeAddition(RECORD1).build();
+    assertTrue(clone.additions().contains(RECORD2));
+    assertFalse(clone.additions().contains(RECORD1));
+    assertTrue(clone.deletions().contains(RECORD3));
+    clone = CHANGE.toBuilder().removeAddition(RECORD2).removeAddition(RECORD1).build();
+    assertFalse(clone.additions().contains(RECORD2));
+    assertFalse(clone.additions().contains(RECORD1));
+    assertTrue(clone.additions().isEmpty());
+    assertTrue(clone.deletions().contains(RECORD3));
+  }
+
+  @Test
+  public void testRemoveDeletion() {
+    ChangeRequest clone = CHANGE.toBuilder().removeDeletion(RECORD3).build();
+    assertFalse(clone.deletions().contains(RECORD3));
```

Comment:
Nit: this line and the one below are slightly redundant; we could probably just choose one of them.